lib/ruby_indexer/lib/ruby_indexer/declaration_listener.rb

```diff
@@ -359,6 +346,28 @@ def on_alias_method_node_enter(node)
 
     private
 
+    sig do
+      params(
+        node: T.any(
+          Prism::InstanceVariableAndWriteNode,
+          Prism::InstanceVariableOperatorWriteNode,
+          Prism::InstanceVariableOrWriteNode,
+          Prism::InstanceVariableTargetNode,
+          Prism::InstanceVariableWriteNode,
+        ),
+        loc: Prism::Location,
+      ).void
+    end
+    def handle_instance_variable(node, loc)
+      name = node.name.to_s
+      return if name == "@"
+
+      # When instance variables are declared inside the class body, they turn into class instance variables rather than
+      # regular instance variables
+      owner = @inside_def ? @owner_stack.last : singleton_klass
+      @index << Entry::InstanceVariable.new(name, @file_path, loc, collect_comments(node), owner)
```

Comment:
Just to confirm if my understanding is correct: the only case where `owner` is `nil` is when the ivars are defined at the top-level, which becomes ivars of the `main` object?